pkg/yamlprocessor/yamlprocessor_test.go

```diff
@@ -253,6 +253,16 @@ foo:
   - baz`),
 			wantErr: false,
 		},
+		{
+			name:    "new line at end of yml given",
+			yml:     `foo: bar
```

Comment:
what if raw yaml has more than one `\n` at the end of file 🤔

Reply:
@khanhtc1202 It's just as you said...

How about simply always adding a new line at the EOF?

According to the POSIX Stndard, The definition of a LINE is
> A sequence of zero or more non- <newline> characters plus a terminating <newline> character.

https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
In that case, I guess diff will be shown if the input file does not contain a new line at EOF, thus the idea of making no diff caused by a new line at EOF is not guaranteed.

I think we should update the logic to count new lines at EOF and add (if necessary) new line(s) to the end of parsed file, wdyt? 👀

Reply:
My suggestion is to always add single newline to the end of the output file, whether or not there are some newlines at the end of the source file.

Reply:
I understand your suggestion 👍 How about changing it to: add a new line to the EOF in case of there is no `\n` at EOF? If we just add a new line anyway (regardless there is a new line at the EOF or not), in case there is already one `\n` at EOF, we added an unnecessary new line and could cause a diff shown, wdyt?

Reply:
I think there is no `\n` at EOF in generated yaml. Because yamlprocessor calls here (https://github.com/goccy/go-yaml/blob/master/ast/ast.go#L610 ).

```
return strings.Join(doc, "\n")
```

Based on this, which of the following should I do?

1. modify yamlprocessor to add single newline to the end of the output file
2. add a new line to the EOF in case of there is no \n at EOF
3. Make Pull Request to `goccy/go-yaml` to append `\n` at EOF
4. Do 2 and 3

Reply:
I think (2) is good enough for pipecd, about (3) may need discuss on `goccy/go-yaml` repo

Reply:
OK, I'll fix it in the way of (2).

And (3) will discuss on https://github.com/goccy/go-yaml/pull/329 .